reflect.go

```diff
@@ -257,6 +256,17 @@ func (r *Reflector) reflectTypeToSchema(definitions Definitions, t reflect.Type)
 		}
 
 	case reflect.Map:
+		var i int
+		if t.Key() == reflect.TypeOf(i) {
```

Comment:
To be consistent this should be extended to cover all of the integer numeric types, and probably floats as well (though the pattern would be quite a bit more complex for floats).

A more convenient way to do this than comparing types is to compare the kind:

```
switch t.Key().Kind() {
  case reflect.Int, reflect.Int8, reflect.Int16, ...
  case reflect.Float32, reflect.Float64:
}